### PR TITLE
Merge data snapshots

### PR DESCRIPTION
**Pull request by ibash**
Fixes https://github.com/goldibex/targaryen/issues/26 by merging root and newData when trying writes.

### CODE REVIEW

File: lib/ruleset.js [hunk 1]
```diff
@@ -278,7 +238,8 @@ Ruleset.prototype.tryWrite = function(path, root, newData, auth, skipWrite, skip
   // write encompasses both the cascading write rules and the
   // non-cascading validate rules
 
-  var newDataRoot = makeNewDataSnap(path, RuleDataSnapshot.convert(newData));
+  var newDataRoot = makeNewDataSnap(path, newData);
+  newDataRoot = root.merge(newDataRoot)
```
Comment (goldibex) [inline lib/ruleset.js]: Can you put these on one line, like so:

```js
var newDataRoot = root.merge(makeNewDataSnap(path, newData));
```

**Comment by goldibex** [comment]
This solves a whole host of problems. Thank you so much for submitting.

Once that one change is made, I'll merge and release.

**Comment by ibash** [comment]
Happy to help :)

**Comment by goldibex** [comment]
LGTM

**Pull request merged**